### PR TITLE
Fixes #9117 - Upgrade secure_headers to version 3.4

### PR DESCRIPTION
**Pull request by tbrisker**
Any plugin that makes changes to secure headers may need to be updated
to correctly handle this upgrade, as there have been some breaking
changes to the secure_headers api.

### CODE REVIEW

File: test/functional/application_controller_subclass_test.rb [hunk 1]
```diff
@@ -135,20 +135,6 @@ class TestableResourcesControllerTest < ActionController::TestCase
     end
   end
 
-  context "secure headers in HTTP response" do
```
Comment (tbrisker) [inline test/functional/application_controller_subclass_test.rb:138]: secure headers are now added via middleware, so are not available to functional tests.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -20,6 +19,7 @@ class ApplicationController < ActionController::Base
   before_action :set_taxonomy, :require_mail, :check_empty_taxonomy
   before_action :authorize
   before_action :welcome, :only => :index, :unless => :api_request?
+  before_action :allow_webpack, :if => proc {Rails.configuration.webpack.dev_server.enabled}
```
Comment (dLobatog) [inline app/controllers/application_controller.rb]: Nitpick, `->` for one liners (I thought rubocop warned about this one)

**Comment by tbrisker** [comment]
@dLobatog addressed your comment and rebased.

**Comment by tbrisker** [comment]
[test]

File: test/integration/middleware_test.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      "img-src 'self' *.gravatar.com; script-src 'unsafe-eval' 'unsafe-inline' "+
+      "'self'; style-src 'unsafe-inline' 'self'"
+  end
+
+  test "webpack dev server adds the dev server to Content-Security-Policy" do
+    Rails.configuration.webpack.dev_server.enabled = true
+    Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+    visit '/'
+    assert page.response_headers['Content-Security-Policy'].include?(Rails.configuration.webpack.dev_server.port.to_s)
+    Rails.configuration.webpack.dev_server.enabled = false
```
Comment (domcleal) [inline test/integration/middleware_test.rb]: Nitpick, please put this line into an `ensure` block - if the test ever fails, it prevents other tests from being affected, making the root cause a lot clearer.

Comment (tbrisker) [reply test/integration/middleware_test.rb]: fixed.

**Comment by domcleal** [comment]
Merged as 5a368d8, thanks @tbrisker.

**Pull request closed without merging**